File: Server/Server.Application/Users/RegisterCustomer/RegisterCustomerCommand.cs
using Server.Application.Abstractions.Messaging;
using Server.Domain.Users;

namespace Server.Application.Users.RegisterCustomer;
 
public sealed record RegisterCustomerCommand(
    string FirstName,
    string LastName,
    string Email,
    string PhoneNumber,
    Gender Gender,
    string Country,
    string City,
    string ZipCode,
    string Street,
    string Password) : ICommand<Guid>;

File: Client/src/entities/users/api/register-customer/register-customer-command.ts
import { Gender } from "../../model";

export type RegisterCustomerCommand = {
	firstName: string;
	lastName: string;
	email: string;
	phoneNumber: string;
	gender: Gender;
	country: string;
	city: string;
	zipCode: string;
	street: string;
	password: string;
};
